nixos/modules/services/networking/adguardhome.nix

```diff
@@ -0,0 +1,97 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.adguardhome;
+
+  args = concatStringsSep " " ([
+    "--no-check-update"
+    "--config ${cfg.confFile}"
+    "--work-dir ${cfg.workDir}"
+    "--host ${cfg.host}"
+    "--port ${toString cfg.port}"
+  ] ++ optional (cfg.logFile != null) "--logfile ${cfg.logFile}"
+  ++ optional (cfg.pidFile != null) "--pidfile ${cfg.pidFile}"
+  ++ optional cfg.verbose "--verbose");
+
+in
+{
+  options.services.adguardhome = with types; {
+    enable = mkEnableOption "AdGuard Home network-wide ad blocker";
+
+    confFile = mkOption {
+      default = "/etc/AdGuardHome/AdGuardHome.yaml";
+      type = path;
+      description = ''
+        Path to the config file.
+      '';
+    };
+
+    workDir = mkOption {
+      default = "/var/lib/AdGuardHome";
+      type = path;
+      description = ''
+        Path to the working directory.
+      '';
+    };
+
+    host = mkOption {
+      default = "0.0.0.0";
+      type = str;
+      description = ''
+        Host address to bind HTTP server on.
+      '';
+    };
+
+    port = mkOption {
+      default = 3000;
+      type = port;
+      description = ''
+        Port to serve HTTP pages on.
+      '';
+    };
+
+    logFile = mkOption {
+      default = "syslog";
+      type = nullOr (either path (enum [ "syslog" ]));
+      description = ''
+        Path to log file. If null: write to stdout; if 'syslog': write to system log.
+      '';
+    };
+
+    pidFile = mkOption {
```

Comment:
Under what scenario does this need to be configurable?

Reply:
`pidFile` can't be specified in the config file and if it wasn't configurable (e.g. fixed to `/run/AdGuardHome/AdGuardHome.pid`) it would be inconvenient to override. A few other modules (mongodb, bitcoind) do allow `pidFile` to be configured, but I admit I can't think of a use case.

Alternatively, the pid file could be disabled by default and enabled when needed through an `extraArgs` option. Could probably at least remove `verbose` in that case too.